### ISSUE TITLE
can buck support Android Data Binding library?

### ISSUE THREAD

**Issue by kissazi2** [facebook/buck#371]
if I use Android data binding library,I must use Android Studio to compile my project.Can I use Buck to compile Data Binding project?

**Comment by sdwilsh** [comment]
We can, but I don't think we'd do it until it's out of beta (since we'll have to reverse engineer how Gradle works with it).

**Comment by kennethmath** [comment]
k

**Comment by skyisle** [comment]
Now, it's out of beta with android build 1.5.0 release.

**Comment by jasta** [comment]
@skyisle I believe this library remains in beta, according to both https://bintray.com/android/android-tools/com.android.databinding.dataBinder/view and https://android.googlesource.com/platform/frameworks/data-binding/+/studio-master-dev

**Comment by yangwuan55** [comment]
Hope to support as soon as possible, please.

**Comment by skyisle** [comment]
@jasta You are right. Only some modules is released such as compiler, base libs not extensions. https://android.googlesource.com/platform/frameworks/data-binding/+/9adb2e8a1fdd7352599ccb16e471c97049f97ae9%5E%21/#F0

**Comment by archmages** [comment]
Android data binding now is beta version, I hope Buck will support it when it update to stable version

**Comment by jasta** [comment]
Databinding appears to be out of beta and is now using version 1.0 and 1.1 as the two latest releases.

**Comment by nbonavia** [comment]
Any feedback on buck supporting Android databinding? It's the only blocker I have to start using buck for my build processes.

**Comment by Piasy** [comment]
In deed Android data binding has already been stable, when could BUCK support them?

**Comment by loveguag** [comment]
Any update？Data binding has been more widely used now~

**Comment by sdwilsh** [comment]
Nothing has changed since March 1st with regards to the core team having a need for this, so no, there are no updates :)

**Comment by jaggs6** [comment]
any updates on this?

**Comment by Coneko** [comment]
None, see @sdwilsh's comment above.

**Comment by jaggs6** [comment]
@Coneko obviously I have read the comment above. It's been three months. Things might have changed.

**Comment by Coneko** [comment]
Sorry I meant this comment:

> I suspect the project would accept a pull request, but nobody on the core team has a need for it so it is unlikely to be done by the core team.

**Comment by jaggs6** [comment]
@sdwilsh any updates?

**Comment by sdwilsh** [comment]
Negative.  As far as I know, this isn't used at Facebook, which makes it unlikely that anybody there is going to add it.  Uber has started to use Buck pretty heavily and contributing quite a bit, but I'm guessing they don't use this either because they've sent zero PRs over about this.

Someone from the community is likely going to need to step up to do this if they care enough about it.

**Comment by connected-jallen** [comment]
@sdwilsh The reason why no one at Facebook is using Data Binding is because Buck does not support Data Binding. The reason you give is a catch 22 :) Data Binding is pretty awesome. It might be better for Android dev in Facebook if Buck kicked off a Gradle job in the end.

**Comment by yangwuan55** [comment]
I think we can use the freeline now.

**Comment by rahul-a** [comment]
I am working on trying to add support for databinding to Buck but I noticed some weird behavior. 

So right now I'm plugging in the annotation processor into my `android_library` which is dependent on an `android_resource` (I've added a step to write intermediate layout info to project's gen path) and processing works fine but due to my library files being `DummyRDotJava` files their declarations are non final and that breaks the output of DataBinding Processor (since it uses the R declarations in a switch case) whereas if I explicitly make them final it gives me a Resource not found exception when running the app. 

Also I noticed that sometimes when my BUCK build file changes for a project and I try to build a lib which has resources, the resources rule is not rebuilt and it's last generated artifacts disappear from buck-out, the problem I face is one of the databinding steps has to depend on the artifacts generated by `android_resource` rule.

Ideas on how to tackle these issues would help a lot.

**Comment by zhaoshuyu** [comment]
buck supports android databinding now ?  thanks

**Comment by ameyab10** [comment]
Oh gosh, total catch 22 here.. Would be awesome if buck could support this since databinding is pretty mainstream now.

**Comment by ajanuar** [comment]
Still no new update on data binding?

**Comment by yangwuan55** [comment]
I don't know who still uses Buck.I think if you use databinding just use gradle,best choice.


> Still no new update on data binding?
